### PR TITLE
Major dependencies update

### PR DESCRIPTION
**Pull request by sainak**
## Proposed Changes

- update docker images to 3.12-slim-bookworm
- update django to 5.1.1 (ran django-upgrade to fix deprecated warnings)
- dropped djdt as it is not compatibe with py312
- update dependabot config to create grouped pull request for dependency updates

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@ohcnetwork/care-backend-maintainers @ohcnetwork/care-backend-admins

### CODE REVIEW

**Review by vigneshhari — APPROVED** [review]

File: docker/prod.Dockerfile [hunk 1]
```diff
@@ -38,44 +41,26 @@ RUN ARCH=$(dpkg --print-architecture) && \
...
-
-WORKDIR ${APP_HOME}
-
 RUN apt-get update && apt-get install --no-install-recommends -y \
   libpq-dev gettext wget curl gnupg \
   && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false \
   && rm -rf /var/lib/apt/lists/*
 
-# copy typst binary from builder stage
-COPY --from=builder --chmod=0755 /usr/local/bin/typst /usr/local/bin/typst
```
Comment (vigneshhari) [inline docker/prod.Dockerfile:72]: Revert, we need typst

File: config/settings/local.py [hunk 2]
```diff
@@ -13,22 +13,6 @@
 # https://github.com/jazzband/django-silk#requirements
 INSTALLED_APPS += ["silk"]  # noqa F405
 MIDDLEWARE += ["silk.middleware.SilkyMiddleware"]  # noqa F405
-# https://github.com/jazzband/django-silk#profiling
-SILKY_PYTHON_PROFILER = True
```
Comment (vigneshhari) [inline config/settings/local.py:17]: We can keep the profiler.

File: config/authentication.py [hunk 3]
```diff
@@ -209,7 +209,7 @@ def authenticate_header(self, request):
         return "Bearer"
 
     def authenticate(self, request):
-        jwt_token = request.META.get("HTTP_AUTHORIZATION")
+        jwt_token = self.get_header(request)
```
Comment (vigneshhari) [inline config/authentication.py:212]: are you sure we get the Auth header with this method? it will default to the header we use in settings.

**Review by vigneshhari — CHANGES_REQUESTED** [review]

File: config/settings/local.py (hunk 2 above)
Comment (sainak) [reply config/settings/local.py:17]: https://github.com/jazzband/django-silk?tab=readme-ov-file#profiling python 312 borked nested profilers

File: config/authentication.py (hunk 3 above)
Comment (sainak) [reply config/authentication.py:212]: yes 
image

**Review by vigneshhari — APPROVED** [review]

**Pull request merged**